src/peer/negotiator.js

```diff
@@ -550,8 +550,13 @@ class Negotiator extends EventEmitter {
    */
   _replacePerStream(newStream) {
     const localStreams = this._pc.getLocalStreams();
+
+    // temporarily unset onnegotiationneeded so that it doesn't do anything on removeStream.
+    const tmpOnNegotiationNeeded = this._pc.onnegotiationneeded;
+    this._pc.onnegotiationneeded = () => {};
+
     // We assume that there is at most 1 stream in localStreams
-    if (localStreams && localStreams[0]) {
+    if (localStreams instanceof Array && localStreams.length > 0) {
```

Comment:
If our target browsers has `Array.isArray()`, we should use this.

```js
const fake = { __proto__: Array.prototype };
fake instanceof Array; // true...
```

Reply:
sorry I don't understand.

are you saying the code should be this?

```javascript
if (Array.isArray(localStreams) && localStreams.length > 0) {
```

what's the advantage of using `isArray()` over `instanceof`?

Reply:
> are you saying the code should be this?

Yes.

> what's the advantage of using isArray() over instanceof?

1. shows in example code,

```js
// this is not an array, this is object
const fake = { __proto__: Array.prototype };

// but this returns true
fake instanceof Array;

// this returns false
Array.isArray({ __proto__: Array.prototype });
```

And `isArray()` is more clear syntax for everyone.

Reply:
Sorry, I notice this now..

Variable `localStreams` comes from `pc.getLocalStreams()`, so it must be an Array.
If there are no streams, it returns empty array(`[]`).

So, I think we can skip type check.

```js
if (localStreams.length > 0) {
```

is enough.

Reply:
fixed